spa_ui/self_service/client/app/states/marketplace/list/list.state.js

```diff
@@ -45,20 +46,37 @@
       $state.go('marketplace.details', {serviceTemplateId: template});
     }
 
+    var categoryNames = [];
+
+    angular.forEach(vm.serviceTemplates, function(item) {
```

Comment:
recommend avoiding inline functions, rather give it a name build it out elsewhere 
```
angular.forEach(vm.serviceTemplates, categoryNameBuilder)

function categoryNameBuilder(item){
...
}